src/plugins/dashboard/public/application/lib/dashboard_panel_storage.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Storage } from '../../services/kibana_utils';
+import { SavedDashboardPanel } from '..';
+
+export const DASHBOARD_PANELS_UNSAVED_ID = 'unsavedDashboard';
+const DASHBOARD_PANELS_SESSION_KEY = 'dashboardStateManagerPanels';
+
+export class DashboardPanelStorage {
+  private sessionStorage: Storage;
+
+  constructor() {
+    this.sessionStorage = new Storage(sessionStorage);
```

Comment:
I did a bit of math in the linked issue - and extrapolating from that, it seems like we can store ~600 maps panels by value in the session storage before running out of memory. 

It may still be worth using the hashed item store, but it may be worth a follow-up PR because it seems like I would have to change the implementation significantly e.g. using the dashboard id and the storage key as a hash.

Am I overestimating the amount of effort required here?

Reply:
Not sure about the effort, but indeed this storage more complicated than we need it to be for the dashboard use case. This storage is used when you enable `state:storeInSessionStorage` to store browser history in a session storage. 

Maybe it still would make sense to wrap `setItem` into try/catch with a toast